components/script/dom/window.rs

```diff
@@ -734,30 +736,43 @@ impl WindowMethods<crate::DomTypeHolder> for Window {
         }
         let (sender, receiver) =
             ProfiledIpc::channel(self.global().time_profiler_chan().clone()).unwrap();
-        let prompt = PromptDefinition::Alert(s.to_string(), sender);
-        let msg = EmbedderMsg::Prompt(self.webview_id(), prompt, PromptOrigin::Untrusted);
+        let dialog = ScriptDialog::Alert {
+            message: s.to_string(),
+            response_sender: sender,
+        };
+        let msg = EmbedderMsg::ShowDialog(self.webview_id(), dialog);
         self.send_to_embedder(msg);
-        receiver.recv().unwrap();
+        let AlertResponse::Ok = receiver.recv().unwrap();
```

Comment:
Just a little bit concerned about the `unwrap()` here. Since the embedder may not handle the sender well, will it be better to log the error instead of panic?

Reply:
with the delegate request pattern in place, libservo will ensure that the embedder always sends a valid response, so this can only happen if ipc is seriously broken. i guess it depends on whether we want to treat that as equivalent to dismissing these dialogs, or if panicking the script thread is ok?

Reply:
Yes, since we are supporting multiple webviews(tabs), panic in the script thread will make all tabs stop working. I think we can discuss this in another thread.

Reply:
btw, I don't have permission to review the issue. 

There are no other questions from my side :)